src/thunderbots/software/ai/hl/stp/evaluation/enemy_threat.cpp

```diff
@@ -0,0 +1,169 @@
+#include "ai/hl/stp/evaluation/enemy_threat.h"
+
+#include <deque>
+
+#include "ai/hl/stp/evaluation/intercept.h"
+#include "ai/hl/stp/evaluation/possession.h"
+#include "ai/hl/stp/evaluation/team.h"
+#include "ai/world/world.h"
+#include "geom/util.h"
+#include "shared/constants.h"
+
+std::optional<std::pair<int, std::optional<Robot>>> Evaluation::getNumPassesToRobot(
+    const Robot& initial_passer, const Robot& final_receiver, const Team& team,
+    const std::vector<Robot>& other_robots)
+{
+    if (initial_passer == final_receiver)
+    {
+        return std::make_pair(0, std::nullopt);
+    }
+
+    // We calculate the minimum number of passes it would take for the initial_passer
+    // robot to pass the ball to the final_receiver, assuming both robots are on the given
+    // team
+    //
+    // This algorithm essentially treats the team of robots like a Directed Acyclic
+    // Graph and "traverses" the graph to find the shortest path to the robot
+
+    // The robots that could have the ball and make a pass at each iteration. They
+    // are on the "frontier" of the graph search
+    std::vector<Robot> current_passers{initial_passer};
+    // The remaining robots we haven't checked yet
+    std::vector<Robot> unvisited_robots = team.getAllRobots();
+    // Remove the robot already in the current_passers since it starts off as visited
+    unvisited_robots.erase(
+        std::remove(unvisited_robots.begin(), unvisited_robots.end(), initial_passer),
+        unvisited_robots.end());
+
+    // On each iteration, check what robots can be passed to. These receivers will
+    // become the passers on the next iteration. This is like expanding the frontier
+    // of the graph
+    //
+    // We continue to iterate and check greater numbers of passes until one of the
+    // following cases:
+    // * There are no more passers at the end of the iteration. This means no more
+    //   unvisited robots can be passed to
+    // * There are no more unvisited robots
+    // * We have iterated up to the size of the team. This is a fallback case to
+    //   prevent any infinite loops, just in case
+    //
+    // The pass_num starts at one since the 0-th pass would be when the robot already
+    // has the ball
```

Comment:
I'm a bit confused about what this last sentence means?

Reply:
We already check the case where the passer and receiver are the same. If this was the case that would require `0` passes. So when we start the loop we are checking for the possibility of the receiver getting the ball in `1` pass. I'll try make the comments more clear.